### ISSUE TITLE
Deny access to unnecessary / sensitive plugin files

### ISSUE THREAD

**Issue by strarsis** [roots/trellis#804]
## Submit a feature request or bug report

- [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/guidelines/blob/master/CONTRIBUTING.md)
- [x] This is a feature request
- [ ] This is a bug report
- [x] This request isn't a duplicate of an [existing issue](https://github.com/roots/trellis/issues)
- [x] I've read the [docs](https://roots.io/trellis/docs) and followed them (if applicable)
- [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/c/trellis) forums

The readme.html file contains information about the WordPress installation and is not used on production, it can be requested from `/wp/readme.html` on a Bedrock (Trellis) installation.

It may be a good idea to block access to it in nginx config set up with Trellis, like it is done here: https://github.com/perusio/wordpress-nginx/blob/master/sites-available/example.com.conf#L60

**Comment by swalkinshaw** [comment]
I'd be fine with this. Feel free to do a PR 👍 

There's more files we'd probably want to block:

* `composer.json`
* `license.txt`

**Comment by strarsis** [comment]
There are also other plugins that store data in uploads/ directory by default.

Other files/folders that are optional or shouldn't even be requestable:
- wp/composer.json
- wp/composer.lock
- wp/license.txt
- wp/readme.html

- app/uploads/sucuri
- app/uploads/updraft-plus
- app/uploads/wp-security-audit-log
(more?)

Many plugins add a `.htaccess` file for denying access - which nginx doesn't use.

**Comment by swalkinshaw** [comment]
We could add a variable of an array of filepaths to block access to and we'd only provide the defaults of `wp/` ones.

**Comment by strarsis** [comment]
An opt-in/opt-out feature could be a nice addition.
Either listing the folders in uploads/ folder that should be accessible to public with all others being blocked or listing the folders that should be blocked with all others being accessible.

**Comment by MWDelaney** [comment]
Is it feasible to, by default, allow access to directories that follow WP's "year" naming scheme and deny access, except from the current site's referrer, to all others? This got me in big trouble today when Formidable Forms cheerfully let me know that its upload directory was protected when, in fact, [it wasn't](https://discourse.roots.io/t/nginx-htaccess-equivalent-to-protect-uploads-directories/9502). 

Blocking access except from by the site itself at least resolved the majority of file and URL injection issues presented. 

I would love to contribute to a PR but this stuff is way over my head :(

**Comment by strarsis** [comment]
Now should the access just be denied or rather handed to the WordPress app so it isn't possible to know whether the file exists or not from the outside?

**Comment by MWDelaney** [comment]
The most wordpressy way to handle it would seem to be custom 404. Google prefers that, too.

**Comment by swalkinshaw** [comment]
We have a whole bunch of common files [blocked by default](https://github.com/roots/trellis/blob/0d4b2611a3ca5b02d47f46290ada7fd439a84f6a/roles/wordpress-setup/templates/wordpress-site.conf.j2#L118-L164) now. Via Nginx includes it's also easy to add any more custom ones.
